### DISCUSSION TITLE
How to write plugins?

### DISCUSSION THREAD

**Discussion by notramo** [lapce/lapce#1717] [Q&A]
It is not obvious where to look for info regarding plugin development.

**Comment by theAkito** [comment]
You've probably already looked at this, but I'm still going to post this, in case someone comes along this issue.

* https://docs.lapce.dev/development/plugin-development
* https://plugins.lapce.dev/

**Reply by notramo** [reply]
The first link is about publishing **already written** plugins.
The second is just a curated list of **already written** plugins.

I am looking for info about **writing new** plugins.
I want to write using the [Zig](https://ziglang.org) language. It's ABI compatible with C. What headers are needed, what functions are available?

**Reply by theAkito** [reply]
Yes, I know that.

Though, if there is no documentation on how to do it, the **already written** plugins can serve as an example.

I would like to know as well, where the documentation is regarding the methods of actually **writing new** plugins.
